### PR TITLE
Begin explanation section of docs

### LINKED ISSUE #2412: [docs] add page on AML concepts
https://github.com/jump-dev/JuMP.jl/pull/2395#discussion_r542924527

It would be good to have a page explaining the concept of an AML, what JuMP does and doesn't do, who should and shouldn't use JuMP, and what alternatives people could consider.

### PR DESCRIPTION
**Pull request by odow**
Closes #2412 

Preview: https://jump.dev/JuMP.jl/previews/PR2504/background/should_i_use/

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -47,7 +50,7 @@ programming.
 ## How the documentation is structured
 
 Having a high-level overview of how this documentation is structured will help
-you know where to look for certain things. There are three main parts:
+you know where to look for certain things. There are three four parts:
```
Comment (rtwalker) [inline docs/src/index.md:53]: replaced "three main" with "three four" instead of "four main"

Comment (odow) [reply docs/src/index.md:53]: Oops! Brain fade...

File: docs/src/background/algebraic_modeling_languages.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+A solver is a software package that incorporates algorithms for finding
+solutions to one or more classes of problem.
+
+For example, GLPK is a solver for linear programming (LP) and mixed integer
+programming (MIP) problems. It incorporates algorithms such as the simplex
+method and the interior-point method.
+
+JuMP currently supports a number of open-source and commercial solvers, which
+can be viewed in the [Supported-solvers](@ref) table.
```
Comment (joaquimg) [inline docs/src/background/algebraic_modeling_languages.md:173]: Perhaps this is a good place to say that solvers are 100% responsible for the quality of the solution.

File: docs/src/background/algebraic_modeling_languages.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+Each mathematical optimization solver API has its own concepts and data
+structures for representing optimization models and obtaining results.
+However, it is often desirable to represent an instance of an optimization
+problem at a higher level so that it is easy to try using different solvers.
+
+MathOptInterface (MOI) is an abstraction layer designed to provide an
+interface to mathematical optimization solvers so that users do not need to
+understand multiple solver-specific APIs. MOI can be used directly, or through
+a higher-level modeling interface like JuMP.
```
Comment (joaquimg) [inline docs/src/background/algebraic_modeling_languages.md:185]: Linking the manual, paper and repo would be nice.

File: docs/src/background/algebraic_modeling_languages.md (hunk 2 above)
Comment (odow) [reply docs/src/background/algebraic_modeling_languages.md:173]: Yip. this part is still very much WIP.

**Pull request merged**